bl/interfaces/controller.go

```diff
@@ -0,0 +1,47 @@
+package interfaces
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/kimbellG/kerror"
+	ttgrpc "github.com/kimbellG/tournament-bl/interfaces/grpc"
+	"github.com/kimbellG/tournament-bl/models"
+	"github.com/kimbellG/tournament-bl/usecases"
+)
+
+type ServiceController struct {
+	userUsecase usecases.UserUsecase
+}
+
+func (sc *ServiceController) SaveUser(ctx context.Context, user *ttgrpc.User) (*ttgrpc.SaveResponse, error) {
+	mUser, err := userGrpcToModels(user)
+	if err != nil {
+		return nil, kerror.Errorf(err, "marshaling user struct to models")
+	}
+
+	id, err := sc.userUsecase.Save(mUser)
+	if err != nil {
+		return nil, kerror.Errorf(err, "save user")
+	}
+
+	return &ttgrpc.SaveResponse{
+		Id: id.String(),
+	}, nil
+}
+
+func userGrpcToModels(gUser *ttgrpc.User) (*models.User, error) {
```

Comment:
right idea, wrong place to put. You can put this function in models package and make it exported. And I'd call it UserFromProto (because technically grpc is just protocol and this User model is generated from protobuf)

Reply:
I don't understand, why should I put this func to  models layer? If I do this, models layer will know about grpc layer. This is violation of principle of dependency injection